go/libraries/doltcore/doltdb/index_editor.go

```diff
@@ -68,6 +68,8 @@ func (indexEd *IndexEditor) Flush(ctx context.Context) error {
 	indexEd.flushMutex.Lock()
 	defer indexEd.flushMutex.Unlock()
 
+	defer indexEd.ed.Close() // current edit accumulator is captured by defer
```

Comment:
Might add a note about why this is important so somebody after you doesn't reintroduce the leak